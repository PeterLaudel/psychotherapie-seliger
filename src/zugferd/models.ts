export interface Position {
  id: string;
  description: string;
  quantity: number;
  price: number;
  tax: number;
}

export interface Seller {
  identifier: string;
  vatId: string;
  name: string;
  street: string;
  city: string;
  zip: string;
  country: string;
}

export interface Buyer {
  name: string;
  street: string;
  city: string;
  zip: string;
  country: string;
}

export interface ZugferdData {
  invoiceNumber: string;
  invoiceDate: string;
  positions: Position[];
  seller: Seller;
  buyer: Buyer;
}
