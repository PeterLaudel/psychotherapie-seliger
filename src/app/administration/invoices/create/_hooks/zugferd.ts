import { create } from "xmlbuilder2";
import { XMLBuilder } from "xmlbuilder2/lib/interfaces";

interface ZugferdPosition {
  id: string;
  description: string;
  quantity: number;
  price: number;
  tax: number;
}

interface ZugferdPartie {
  name: string;
  street: string;
  city: string;
  zip: string;
  country: string;
}

export interface InvoiceData {
  number: string;
  date: string;
  positions: ZugferdPosition[];
  seller: ZugferdPartie;
  buyer: ZugferdPartie;
}

export function createZugferdXml(invoiceData: InvoiceData): string {
  const { positions, number, date, seller, buyer } = invoiceData;

  // prettier-ignore
  const xmlDoc = create({ version: "1.0", encoding: "UTF-8" }).ele(
    "rsm:CrossIndustryInvoice",
    {
      "xmlns:rsm": "urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100",
      "xmlns:ram": "urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100",
      "xmlns:udt": "urn:un:unece:uncefact:data:standard:UnqualifiedDataType:100",
    }
  );

  // prettier-ignore
  xmlDoc.ele("rsm:ExchangedDocumentContext")
    .ele("ram:GuidelineSpecifiedDocumentContextParameter")
      .ele("ram:ID").txt("urn:factur-x:en16931:basic").up()
    .up()
  .up();

  // prettier-ignore
  xmlDoc.ele("rsm:ExchangedDocument")
    .ele("ram:ID").txt(number).up()
    .ele("ram:TypeCode").txt("380").up() // 380 = Commercial invoice
    .ele("ram:IssueDateTime")
      .ele("udt:DateTimeString", { format: "102" }).txt(date).up()
    .up()
  .up();

  const transaction = xmlDoc.ele("rsm:SupplyChainTradeTransaction");

  // Add all line items first
  addPositions(transaction, positions);

  // Add ApplicableHeaderTradeAgreement with Seller and Buyer
  const agreement = transaction.ele("ram:ApplicableHeaderTradeAgreement");
  agreement.ele("ram:SellerTradeParty")
    .ele("ram:Name").txt(seller.name).up()
    .ele("ram:PostalTradeAddress")
      .ele("ram:PostcodeCode").txt(seller.zip).up()
      .ele("ram:LineOne").txt(seller.street).up()
      .ele("ram:CityName").txt(seller.city).up()
      .ele("ram:CountryID").txt(seller.country).up()
    .up()
  .up();
  agreement.ele("ram:BuyerTradeParty")
    .ele("ram:Name").txt(buyer.name).up()
    .ele("ram:PostalTradeAddress")
      .ele("ram:PostcodeCode").txt(buyer.zip).up()
      .ele("ram:LineOne").txt(buyer.street).up()
      .ele("ram:CityName").txt(buyer.city).up()
      .ele("ram:CountryID").txt(buyer.country).up()
    .up()
  .up();

  // Add empty ApplicableHeaderTradeDelivery (required by schema)
  transaction.ele("ram:ApplicableHeaderTradeDelivery");

  // Add ApplicableHeaderTradeSettlement
  addMonetarySummary(transaction, positions);

  return xmlDoc.end({ prettyPrint: true });
}

function addPositions(xmlDoc: XMLBuilder, positions: ZugferdPosition[]): void {
  positions.forEach((position) => {
    const lineItem = xmlDoc.ele("ram:IncludedSupplyChainTradeLineItem");
    lineItem
      .ele("ram:AssociatedDocumentLineDocument")
        .ele("ram:LineID")
        .txt(position.id)
        .up()
      .up();
    lineItem
      .ele("ram:SpecifiedTradeProduct")
        .ele("ram:Name")
        .txt(position.description)
        .up()
      .up();
    // SpecifiedLineTradeAgreement: GrossPrice, NetPrice (required)
    const agreement = lineItem.ele("ram:SpecifiedLineTradeAgreement");
    agreement
      .ele("ram:GrossPriceProductTradePrice")
        .ele("ram:ChargeAmount")
        .txt(position.price.toFixed(2))
        .up()
      .up();
    agreement
      .ele("ram:NetPriceProductTradePrice")
        .ele("ram:ChargeAmount")
        .txt(position.price.toFixed(2))
        .up()
      .up();

    lineItem
      .ele("ram:SpecifiedLineTradeDelivery")
        .ele("ram:BilledQuantity", { unitCode: "C62" })
        .txt(position.quantity.toString())
        .up()
      .up();

    // SpecifiedLineTradeSettlement: ApplicableTradeTax, SpecifiedTradeSettlementLineMonetarySummation
    const settlement = lineItem.ele("ram:SpecifiedLineTradeSettlement");
    const tax = settlement.ele("ram:ApplicableTradeTax");
    tax.ele("ram:TypeCode").txt("VAT").up();
    tax.ele("ram:CategoryCode").txt("E").up();
    tax.ele("ram:ExemptionReasonCode").txt("E").up();
    tax.ele("ram:RateApplicablePercent").txt(position.tax.toFixed(2)).up();
    settlement
      .ele("ram:SpecifiedTradeSettlementLineMonetarySummation")
        .ele("ram:LineTotalAmount")
        .txt((position.price * position.quantity).toFixed(2))
        .up()
      .up();
  });
}

function addMonetarySummary(
  xmlDoc: XMLBuilder,
  positions: ZugferdPosition[]
): void {
  const totalAmount = positions.reduce(
    (sum, p) => sum + p.price * p.quantity,
    0
  );

  const settlement = xmlDoc.ele("ram:ApplicableHeaderTradeSettlement");

  // Add required InvoiceCurrencyCode before payment means
  settlement.ele("ram:InvoiceCurrencyCode").txt("EUR").up();

  // prettier-ignore
  settlement.ele("ram:SpecifiedTradeSettlementPaymentMeans")
    .ele("ram:TypeCode").txt("42").up() // 42 = Bank transfer
    .up();

  // Header-level tax: order and content per schema
  const tax = settlement.ele("ram:ApplicableTradeTax");
  tax.ele("ram:CalculatedAmount").txt("0.00").up();
  tax.ele("ram:TypeCode").txt("VAT").up();
  tax.ele("ram:ExemptionReason").txt("Medical service").up();
  tax.ele("ram:BasisAmount").txt(totalAmount.toFixed(2)).up();
  tax.ele("ram:CategoryCode").txt("E").up();
  tax.ele("ram:ExemptionReasonCode").txt("E").up();
  tax.ele("ram:RateApplicablePercent").txt("0.00").up();

  // prettier-ignore
  settlement.ele("ram:SpecifiedTradeSettlementHeaderMonetarySummation")
    .ele("ram:LineTotalAmount").txt(totalAmount.toFixed(2)).up()
    .ele("ram:TaxBasisTotalAmount").txt(totalAmount.toFixed(2)).up()
    .ele("ram:TaxTotalAmount").txt("0.00").up()
    .ele("ram:GrandTotalAmount").txt(totalAmount.toFixed(2)).up()
    .ele("ram:DuePayableAmount").txt(totalAmount.toFixed(2)).up()
    .up();

  settlement.up();
}
