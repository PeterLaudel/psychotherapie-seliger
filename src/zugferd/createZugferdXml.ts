import { create } from "xmlbuilder2";
import { XMLBuilder } from "xmlbuilder2/lib/interfaces";
import type { ZugferdData, Position } from "./models";

export function createZugferdXml(invoiceData: ZugferdData): string {
  const { positions, invoiceNumber, invoiceDate, seller, buyer } = invoiceData;

  const xmlDoc = create({ version: "1.0", encoding: "UTF-8" });
  // prettier-ignore
  const root= xmlDoc.ele(
    "rsm:CrossIndustryInvoice",
    {
      "xmlns:rsm": "urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100",
      "xmlns:ram": "urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100",
      "xmlns:udt": "urn:un:unece:uncefact:data:standard:UnqualifiedDataType:100",
    }
  );

  // prettier-ignore
  root.ele("rsm:ExchangedDocumentContext")
    .ele("ram:GuidelineSpecifiedDocumentContextParameter")
      .ele("ram:ID").txt("urn:cen.eu:en16931:2017").up()
    .up()
  .up();

  // prettier-ignore
  root.ele("rsm:ExchangedDocument")
    .ele("ram:ID").txt(invoiceNumber).up()
    .ele("ram:TypeCode").txt("380").up() // 380 = Commercial invoice
    .ele("ram:IssueDateTime")
      .ele("udt:DateTimeString", { format: "102" }).txt(invoiceDate).up()
    .up()
  .up();

  const transaction = root.ele("rsm:SupplyChainTradeTransaction");

  // Add all line items first
  addPositions(transaction, positions);

  // Add ApplicableHeaderTradeAgreement with Seller and Buyer
  const agreement = transaction.ele("ram:ApplicableHeaderTradeAgreement");

  // prettier-ignore
  agreement.ele("ram:SellerTradeParty")
    .ele("ram:Name").txt(seller.name).up()
    .ele("ram:PostalTradeAddress")
      .ele("ram:PostcodeCode").txt(seller.zip).up()
      .ele("ram:LineOne").txt(seller.street).up()
      .ele("ram:CityName").txt(seller.city).up()
      .ele("ram:CountryID").txt(seller.country).up()
    .up()
    .ele("ram:SpecifiedTaxRegistration")
      .ele("ram:ID", { schemeID: "VA" }).txt(seller.vatId).up()
    .up()
  .up();

  // prettier-ignore
  agreement.ele("ram:BuyerTradeParty")
    .ele("ram:Name").txt(buyer.name).up()
    .ele("ram:PostalTradeAddress")
      .ele("ram:PostcodeCode").txt(buyer.zip).up()
      .ele("ram:LineOne").txt(buyer.street).up()
      .ele("ram:CityName").txt(buyer.city).up()
      .ele("ram:CountryID").txt(buyer.country).up()
    .up()
  .up();

  // Add empty ApplicableHeaderTradeDelivery (required by schema)
  transaction.ele("ram:ApplicableHeaderTradeDelivery");

  // Add ApplicableHeaderTradeSettlement
  addMonetarySummary(transaction, positions);

  return root.end({ prettyPrint: true });
}

function addPositions(xmlDoc: XMLBuilder, positions: Position[]): void {
  positions.forEach((position) => {
    const lineItem = xmlDoc.ele("ram:IncludedSupplyChainTradeLineItem");

    // prettier-ignore
    lineItem
      .ele("ram:AssociatedDocumentLineDocument")
        .ele("ram:LineID")
        .txt(position.id)
        .up()
      .up();

    // prettier-ignore
    lineItem
      .ele("ram:SpecifiedTradeProduct")
        .ele("ram:Name")
        .txt(position.description)
        .up()
      .up();
    // SpecifiedLineTradeAgreement: GrossPrice, NetPrice (required)
    const agreement = lineItem.ele("ram:SpecifiedLineTradeAgreement");

    // prettier-ignore
    agreement
      .ele("ram:GrossPriceProductTradePrice")
        .ele("ram:ChargeAmount")
        .txt(position.price.toFixed(2))
        .up()
      .up();

    // prettier-ignore
    agreement
      .ele("ram:NetPriceProductTradePrice")
        .ele("ram:ChargeAmount")
        .txt(position.price.toFixed(2))
        .up()
      .up();

    // prettier-ignore
    lineItem
      .ele("ram:SpecifiedLineTradeDelivery")
        .ele("ram:BilledQuantity", { unitCode: "C62" })
        .txt(position.quantity.toString())
        .up()
      .up();

    // prettier-ignore
    const settlement = lineItem.ele("ram:SpecifiedLineTradeSettlement");
    const tax = settlement.ele("ram:ApplicableTradeTax");
    tax.ele("ram:TypeCode").txt("VAT").up();
    tax.ele("ram:CategoryCode").txt("E").up();
    tax.ele("ram:RateApplicablePercent").txt(position.tax.toFixed(2)).up();
    settlement
      .ele("ram:SpecifiedTradeSettlementLineMonetarySummation")
      .ele("ram:LineTotalAmount")
      .txt((position.price * position.quantity).toFixed(2))
      .up()
      .up();
  });
}

function addMonetarySummary(xmlDoc: XMLBuilder, positions: Position[]): void {
  const totalAmount = positions.reduce(
    (sum, p) => sum + p.price * p.quantity,
    0
  );

  const settlement = xmlDoc.ele("ram:ApplicableHeaderTradeSettlement");

  // Add required InvoiceCurrencyCode before payment means
  settlement.ele("ram:InvoiceCurrencyCode").txt("EUR").up();

  // prettier-ignore
  settlement.ele("ram:SpecifiedTradeSettlementPaymentMeans")
    .ele("ram:TypeCode").txt("42").up() // 42 = Bank transfer
    .up();

  // Header-level tax: order and content per schema
  // prettier-ignore
  const tax = settlement.ele("ram:ApplicableTradeTax");
  tax.ele("ram:CalculatedAmount").txt("0.00").up();
  tax.ele("ram:TypeCode").txt("VAT").up();
  tax.ele("ram:ExemptionReason").txt("Medical service").up();
  tax.ele("ram:BasisAmount").txt(totalAmount.toFixed(2)).up();
  tax.ele("ram:CategoryCode").txt("E").up();
  tax.ele("ram:ExemptionReasonCode").txt("VATEX-EU-132").up();
  tax.ele("ram:RateApplicablePercent").txt("0.00").up();

  // Move SpecifiedTradePaymentTerms BEFORE SpecifiedTradeSettlementHeaderMonetarySummation
  settlement
    .ele("ram:SpecifiedTradePaymentTerms")
    .ele("ram:Description")
    .txt("Due within 30 days")
    .up()
    .ele("ram:DueDateDateTime")
    .ele("udt:DateTimeString", { format: "102" })
    .txt("20231031")
    .up()
    .up()
    .up();

  // prettier-ignore
  settlement.ele("ram:SpecifiedTradeSettlementHeaderMonetarySummation")
    .ele("ram:LineTotalAmount").txt(totalAmount.toFixed(2)).up()
    .ele("ram:TaxBasisTotalAmount").txt(totalAmount.toFixed(2)).up()
    .ele("ram:TaxTotalAmount").txt("0.00").up()
    .ele("ram:GrandTotalAmount").txt(totalAmount.toFixed(2)).up()
    .ele("ram:DuePayableAmount").txt(totalAmount.toFixed(2)).up()
    .up();
}
