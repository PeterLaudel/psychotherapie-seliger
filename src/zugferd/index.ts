export type { ZugferdData, Position, Seller, Buyer } from "./models";
export { createZugferdXml } from "./createZugferdXml";
