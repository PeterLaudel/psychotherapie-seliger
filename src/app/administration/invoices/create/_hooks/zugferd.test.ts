import { createZugferdXml, InvoiceData } from "./zugferd";

describe("createZugferdXml", () => {
  it("should create a valid Zugferd XML document", () => {
    const invoiceData: InvoiceData = {
        number: "INV-12345",
        date: "2023-10-01",
        positions: [
          {
            description: "Therapy session",
            quantity: 1,
            price: 150.0,
            tax: 0.0,
            id: "pos-1",
          },
          {
            description: "Follow-up consultation",
            quantity: 1,
            price: 100.0,
            tax: 0.0,
            id: "pos-2",
          },
        ],
        seller: {
          name: "Therapist Name",
          street: "Therapist Street 123",
          city: "Therapist City",
          zip: "12345",
          country: "Germany",
        },
        buyer: {
          name: "Patient Name",
          street: "Patient Street 456",
          city: "Patient City",
          zip: "67890",
          country: "Germany",
        },
      };

    const xml = createZugferdXml(invoiceData);
    expect(xml).toMatchSnapshot();
  });
});
