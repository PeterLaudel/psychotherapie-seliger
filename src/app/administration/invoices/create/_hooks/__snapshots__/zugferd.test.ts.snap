// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createZugferdXml should create a valid Zugferd XML document 1`] = `
"<?xml version="1.0" encoding="UTF-8"?>
<rsm:CrossIndustryInvoice xmlns:rsm="urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100" xmlns:ram="urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100" xmlns:udt="urn:un:unece:uncefact:data:standard:UnqualifiedDataType:100">
  <rsm:ExchangedDocumentContext>
    <ram:GuidelineSpecifiedDocumentContextParameter>
      <ram:ID>urn:factur-x:en16931:basic</ram:ID>
    </ram:GuidelineSpecifiedDocumentContextParameter>
  </rsm:ExchangedDocumentContext>
  <rsm:ExchangedDocument>
    <ram:ID>INV-12345</ram:ID>
    <ram:TypeCode>380</ram:TypeCode>
    <ram:IssueDateTime>
      <udt:DateTimeString format="102">2023-10-01</udt:DateTimeString>
    </ram:IssueDateTime>
  </rsm:ExchangedDocument>
  <rsm:SupplyChainTradeTransaction>
    <ram:IncludedSupplyChainTradeLineItem>
      <ram:AssociatedDocumentLineDocument>
        <ram:LineID>pos-1</ram:LineID>
      </ram:AssociatedDocumentLineDocument>
      <ram:SpecifiedTradeProduct>
        <ram:Name>Therapy session</ram:Name>
      </ram:SpecifiedTradeProduct>
      <ram:SpecifiedLineTradeAgreement>
        <ram:GrossPriceProductTradePrice>
          <ram:ChargeAmount>150.00</ram:ChargeAmount>
        </ram:GrossPriceProductTradePrice>
        <ram:NetPriceProductTradePrice>
          <ram:ChargeAmount>150.00</ram:ChargeAmount>
        </ram:NetPriceProductTradePrice>
      </ram:SpecifiedLineTradeAgreement>
      <ram:SpecifiedLineTradeDelivery>
        <ram:BilledQuantity unitCode="C62">1</ram:BilledQuantity>
      </ram:SpecifiedLineTradeDelivery>
      <ram:SpecifiedLineTradeSettlement>
        <ram:ApplicableTradeTax>
          <ram:TypeCode>VAT</ram:TypeCode>
          <ram:CategoryCode>E</ram:CategoryCode>
          <ram:ExemptionReasonCode>E</ram:ExemptionReasonCode>
          <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
        </ram:ApplicableTradeTax>
        <ram:SpecifiedTradeSettlementLineMonetarySummation>
          <ram:LineTotalAmount>150.00</ram:LineTotalAmount>
        </ram:SpecifiedTradeSettlementLineMonetarySummation>
      </ram:SpecifiedLineTradeSettlement>
    </ram:IncludedSupplyChainTradeLineItem>
    <ram:IncludedSupplyChainTradeLineItem>
      <ram:AssociatedDocumentLineDocument>
        <ram:LineID>pos-2</ram:LineID>
      </ram:AssociatedDocumentLineDocument>
      <ram:SpecifiedTradeProduct>
        <ram:Name>Follow-up consultation</ram:Name>
      </ram:SpecifiedTradeProduct>
      <ram:SpecifiedLineTradeAgreement>
        <ram:GrossPriceProductTradePrice>
          <ram:ChargeAmount>100.00</ram:ChargeAmount>
        </ram:GrossPriceProductTradePrice>
        <ram:NetPriceProductTradePrice>
          <ram:ChargeAmount>100.00</ram:ChargeAmount>
        </ram:NetPriceProductTradePrice>
      </ram:SpecifiedLineTradeAgreement>
      <ram:SpecifiedLineTradeDelivery>
        <ram:BilledQuantity unitCode="C62">1</ram:BilledQuantity>
      </ram:SpecifiedLineTradeDelivery>
      <ram:SpecifiedLineTradeSettlement>
        <ram:ApplicableTradeTax>
          <ram:TypeCode>VAT</ram:TypeCode>
          <ram:CategoryCode>E</ram:CategoryCode>
          <ram:ExemptionReasonCode>E</ram:ExemptionReasonCode>
          <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
        </ram:ApplicableTradeTax>
        <ram:SpecifiedTradeSettlementLineMonetarySummation>
          <ram:LineTotalAmount>100.00</ram:LineTotalAmount>
        </ram:SpecifiedTradeSettlementLineMonetarySummation>
      </ram:SpecifiedLineTradeSettlement>
    </ram:IncludedSupplyChainTradeLineItem>
    <ram:ApplicableHeaderTradeAgreement>
      <ram:SellerTradeParty>
        <ram:Name>Therapist Name</ram:Name>
        <ram:PostalTradeAddress>
          <ram:PostcodeCode>12345</ram:PostcodeCode>
          <ram:LineOne>Therapist Street 123</ram:LineOne>
          <ram:CityName>Therapist City</ram:CityName>
          <ram:CountryID>Germany</ram:CountryID>
        </ram:PostalTradeAddress>
      </ram:SellerTradeParty>
      <ram:BuyerTradeParty>
        <ram:Name>Patient Name</ram:Name>
        <ram:PostalTradeAddress>
          <ram:PostcodeCode>67890</ram:PostcodeCode>
          <ram:LineOne>Patient Street 456</ram:LineOne>
          <ram:CityName>Patient City</ram:CityName>
          <ram:CountryID>Germany</ram:CountryID>
        </ram:PostalTradeAddress>
      </ram:BuyerTradeParty>
    </ram:ApplicableHeaderTradeAgreement>
    <ram:ApplicableHeaderTradeDelivery/>
    <ram:ApplicableHeaderTradeSettlement>
      <ram:InvoiceCurrencyCode>EUR</ram:InvoiceCurrencyCode>
      <ram:SpecifiedTradeSettlementPaymentMeans>
        <ram:TypeCode>42</ram:TypeCode>
      </ram:SpecifiedTradeSettlementPaymentMeans>
      <ram:ApplicableTradeTax>
        <ram:CalculatedAmount>0.00</ram:CalculatedAmount>
        <ram:TypeCode>VAT</ram:TypeCode>
        <ram:ExemptionReason>Medical service</ram:ExemptionReason>
        <ram:BasisAmount>250.00</ram:BasisAmount>
        <ram:CategoryCode>E</ram:CategoryCode>
        <ram:ExemptionReasonCode>E</ram:ExemptionReasonCode>
        <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
      </ram:ApplicableTradeTax>
      <ram:SpecifiedTradeSettlementHeaderMonetarySummation>
        <ram:LineTotalAmount>250.00</ram:LineTotalAmount>
        <ram:TaxBasisTotalAmount>250.00</ram:TaxBasisTotalAmount>
        <ram:TaxTotalAmount>0.00</ram:TaxTotalAmount>
        <ram:GrandTotalAmount>250.00</ram:GrandTotalAmount>
        <ram:DuePayableAmount>250.00</ram:DuePayableAmount>
      </ram:SpecifiedTradeSettlementHeaderMonetarySummation>
    </ram:ApplicableHeaderTradeSettlement>
  </rsm:SupplyChainTradeTransaction>
</rsm:CrossIndustryInvoice>"
`;
